### ISSUE TITLE
`conda update --all` does not reliably update conda itself in 4.3.x

### ISSUE THREAD

**Issue by ma-ba** [conda/conda#5161]
Conda 4.3.x does not include updates of itself *most of the time* (i.e., unreliably) when running `conda update --all`.
This seems to be a regression introduced in 4.3.0 since I could not reproduce this for 4.2.16.
The behavior is (nearly consistently) reproducible for 4.3.0, 4.3.11 and 4.3.16 (when trying to update to 4.3.17 on `conda-canary`).

Using a fresh Miniconda3 installation:
```bash
$  source miniconda3/bin/activate 
(root) $  conda update --all
Fetching package metadata .........
Solving package specifications: .

Package plan for installation in environment /home/maba/code/miniconda3:

The following packages will be UPDATED:

    openssl:  1.0.2k-0      --> 1.0.2k-1     
    pyasn1:   0.1.9-py36_0  --> 0.2.3-py36_0 
    pycosat:  0.6.1-py36_1  --> 0.6.2-py36_0 
    python:   3.6.0-0       --> 3.6.1-0      
    requests: 2.12.4-py36_0 --> 2.13.0-py36_0

Proceed ([y]/n)? 

openssl-1.0.2k 100% |###############################################################################################################################################################################################################################################################################| Time: 0:00:03 930.70 kB/s
python-3.6.1-0 100% |###############################################################################################################################################################################################################################################################################| Time: 0:00:17   1.01 MB/s
pyasn1-0.2.3-p 100% |###############################################################################################################################################################################################################################################################################| Time: 0:00:00 965.13 kB/s
pycosat-0.6.2- 100% |###############################################################################################################################################################################################################################################################################| Time: 0:00:00   1.02 MB/s
requests-2.13. 100% |###############################################################################################################################################################################################################################################################################| Time: 0:00:00   1.01 MB/s
(root) $  conda update --all
Fetching package metadata .........
Solving package specifications: .

# All requested packages already installed.
# packages in environment at /home/maba/code/miniconda3:
#

```
Updating conda on its own works reliably.
```bash
(root) $  conda update --dry-run conda
Fetching package metadata .........
Solving package specifications: .

Package plan for installation in environment /home/maba/code/miniconda3:

The following packages will be UPDATED:

    conda: 4.3.11-py36_0 --> 4.3.16-py36_0

```
See the following debug outputs:

Failed update:
```bash
(root) $  conda update --debug --dry-run --all
DEBUG conda.core.index:fetch_index(547): channel_urls=OrderedDict([('https://repo.continuum.io/pkgs/free/linux-64', ('defaults', 0)), ('https://repo.continuum.io/pkgs/free/noarch', ('defaults', 0)), ('https://repo.continuum.io/pkgs/r/linux-64', ('defaults', 0)), ('https://repo.continuum.io/pkgs/r/noarch', ('defaults', 0)), ('https://repo.continuum.io/pkgs/pro/linux-64', ('defaults', 0)), ('https://repo.continuum.io/pkgs/pro/noarch', ('defaults', 0))])
Fetching package metadata ...DEBUG requests.packages.urllib3.util.retry:from_int(191): Converted retries value: 3 -> Retry(total=3, connect=None, read=None, redirect=None)
DEBUG conda.core.index:fetch_repodata(481): Using cached repodata for https://repo.continuum.io/pkgs/free/linux-64 at /home/maba/code/miniconda3/pkgs/cache/db552c1e.json. Timeout in 6 sec
DEBUG conda.core.index:read_pickled_repodata(376): found pickle file /home/maba/code/miniconda3/pkgs/cache/db552c1e.q
.DEBUG conda.core.index:fetch_repodata(481): Using cached repodata for https://repo.continuum.io/pkgs/free/noarch at /home/maba/code/miniconda3/pkgs/cache/9ca791dd.json. Timeout in 6 sec
DEBUG conda.core.index:read_pickled_repodata(376): found pickle file /home/maba/code/miniconda3/pkgs/cache/9ca791dd.q
.DEBUG conda.core.index:fetch_repodata(481): Using cached repodata for https://repo.continuum.io/pkgs/r/linux-64 at /home/maba/code/miniconda3/pkgs/cache/6bfb14bd.json. Timeout in 25 sec
DEBUG conda.core.index:read_pickled_repodata(376): found pickle file /home/maba/code/miniconda3/pkgs/cache/6bfb14bd.q
.DEBUG conda.core.index:fetch_repodata(481): Using cached repodata for https://repo.continuum.io/pkgs/r/noarch at /home/maba/code/miniconda3/pkgs/cache/716b88c0.json. Timeout in 25 sec
.DEBUG conda.core.index:fetch_repodata(481): Using cached repodata for https://repo.continuum.io/pkgs/pro/linux-64 at /home/maba/code/miniconda3/pkgs/cache/8db5e652.json. Timeout in 25 sec
DEBUG conda.core.index:read_pickled_repodata(376): found pickle file /home/maba/code/miniconda3/pkgs/cache/8db5e652.q
.DEBUG conda.core.index:fetch_repodata(481): Using cached repodata for https://repo.continuum.io/pkgs/pro/noarch at /home/maba/code/miniconda3/pkgs/cache/08159f17.json. Timeout in 25 sec
.
DEBUG conda.plan:add_defaults_to_specs(391): H0 specs=['readline', 'cryptography', 'idna', 'pip', 'sqlite', 'requests', 'wheel', 'pyasn1', 'yaml', 'tk', 'python', 'openssl', 'cffi', 'pycosat', 'conda-env', 'setuptools', 'pyopenssl', 'libffi', 'pycparser', 'six', 'zlib', 'ruamel_yaml', 'xz', 'conda-env']
DEBUG conda.plan:add_defaults_to_specs(406): H2 python True
DEBUG conda.plan:add_defaults_to_specs(422): H3 python
DEBUG conda.plan:add_defaults_to_specs(406): H2 lua False
DEBUG conda.plan:add_defaults_to_specs(411): H2A lua
DEBUG conda.plan:add_defaults_to_specs(437): HF specs=['readline', 'cryptography', 'idna', 'pip', 'sqlite', 'requests', 'wheel', 'pyasn1', 'yaml', 'tk', 'python', 'openssl', 'cffi', 'pycosat', 'conda-env', 'setuptools', 'pyopenssl', 'libffi', 'pycparser', 'six', 'zlib', 'ruamel_yaml', 'xz', 'conda-env', 'python 3.6* (target=defaults::python-3.6.1-0)']
DEBUG conda.plan:augment_specs(711): Pinned specs=[]
DEBUG conda.resolve:install_specs(794): Checking satisfiability of current install
DEBUG conda.resolve:bad_installed(744): Checking if the current environment is consistent
Solving package specifications: DEBUG conda.resolve:solve(849): Solving for [MatchSpec('readline'), MatchSpec('cryptography'), MatchSpec('idna'), MatchSpec('pip'), MatchSpec('sqlite'), MatchSpec('requests'), MatchSpec('wheel'), MatchSpec('pyasn1'), MatchSpec('yaml'), MatchSpec('tk'), MatchSpec('python'), MatchSpec('openssl'), MatchSpec('cffi'), MatchSpec('pycosat'), MatchSpec('conda-env'), MatchSpec('setuptools'), MatchSpec('pyopenssl'), MatchSpec('libffi'), MatchSpec('pycparser'), MatchSpec('six'), MatchSpec('zlib'), MatchSpec('ruamel_yaml'), MatchSpec('xz'), MatchSpec('conda-env'), MatchSpec('python 3.6* (target=defaults::python-3.6.1-0)'), MatchSpec('conda (target=defaults::conda-4.3.11-py36_0)')]
DEBUG conda.resolve:get_reduced_index(380): Retrieving packages for: [MatchSpec('readline'), MatchSpec('cryptography'), MatchSpec('idna'), MatchSpec('pip'), MatchSpec('sqlite'), MatchSpec('requests'), MatchSpec('wheel'), MatchSpec('pyasn1'), MatchSpec('yaml'), MatchSpec('tk'), MatchSpec('python'), MatchSpec('openssl'), MatchSpec('cffi'), MatchSpec('pycosat'), MatchSpec('conda-env'), MatchSpec('setuptools'), MatchSpec('pyopenssl'), MatchSpec('libffi'), MatchSpec('pycparser'), MatchSpec('six'), MatchSpec('zlib'), MatchSpec('ruamel_yaml'), MatchSpec('xz'), MatchSpec('conda-env'), MatchSpec('python 3.6* (target=defaults::python-3.6.1-0)'), MatchSpec('conda (target=defaults::conda-4.3.11-py36_0)')]
DEBUG conda.resolve:filter_group(405): python: pruned from 82 -> 81
DEBUG conda.resolve:filter_group(405): python: pruned from 81 -> 72
DEBUG conda.resolve:filter_group(405): pip: pruned from 105 -> 86
DEBUG conda.resolve:filter_group(405): python: pruned from 72 -> 2
DEBUG conda.resolve:filter_group(405): openssl: pruned from 22 -> 10
DEBUG conda.resolve:filter_group(405): sqlite: pruned from 5 -> 1
DEBUG conda.resolve:filter_group(405): tk: pruned from 3 -> 1
DEBUG conda.resolve:filter_group(405): xz: pruned from 4 -> 2
DEBUG conda.resolve:filter_group(405): pip: pruned from 86 -> 1
DEBUG conda.resolve:filter_group(405): setuptools: pruned from 198 -> 1
DEBUG conda.resolve:filter_group(405): wheel: pruned from 17 -> 1
DEBUG conda.resolve:filter_group(405): conda-env: pruned from 64 -> 1
DEBUG conda.resolve:filter_group(405): ruamel_yaml: pruned from 10 -> 1
DEBUG conda.resolve:filter_group(405): yaml: pruned from 2 -> 1
DEBUG conda.resolve:filter_group(405): six: pruned from 38 -> 1
DEBUG conda.resolve:filter_group(405): pycparser: pruned from 30 -> 1
DEBUG conda.resolve:filter_group(405): pyopenssl: pruned from 26 -> 1
DEBUG conda.resolve:filter_group(405): cryptography: pruned from 70 -> 1
DEBUG conda.resolve:filter_group(405): cffi: pruned from 68 -> 1
DEBUG conda.resolve:filter_group(405): libffi: pruned from 3 -> 2
DEBUG conda.resolve:filter_group(405): idna: pruned from 13 -> 2
DEBUG conda.resolve:filter_group(405): pyasn1: pruned from 20 -> 2
DEBUG conda.resolve:filter_group(405): pycosat: pruned from 17 -> 2
DEBUG conda.resolve:filter_group(405): requests: pruned from 106 -> 2
DEBUG conda.resolve:filter_group(405): conda: pruned from 378 -> 10
.DEBUG conda.logic:minimize(485): Empty objective, trivial solution
DEBUG conda.resolve:solve(889): Package removal metric: 0
DEBUG conda.logic:minimize(557): Final peak objective: 0
DEBUG conda.logic:minimize(557): Final peak objective: 0
DEBUG conda.resolve:solve(895): Initial package version/build metrics: 0/0
DEBUG conda.logic:minimize(485): Empty objective, trivial solution
DEBUG conda.resolve:solve(900): Track feature count: 0
DEBUG conda.logic:minimize(485): Empty objective, trivial solution
DEBUG conda.resolve:solve(906): Package feature count: 0
DEBUG conda.logic:minimize(557): Final sum objective: 0
DEBUG conda.resolve:solve(911): Dependency update count: 0
DEBUG conda.logic:minimize(557): Final peak objective: 0
DEBUG conda.logic:minimize(485): Empty objective, trivial solution
DEBUG conda.resolve:solve(917): Additional package version/build metrics: 0/0
DEBUG conda.logic:minimize(485): Empty objective, trivial solution
DEBUG conda.resolve:solve(922): Weak dependency count: 0
DEBUG conda.resolve:solve(927): Looking for alternate solutions


# All requested packages already installed.
# packages in environment at /home/maba/code/miniconda3:
#
DEBUG conda.cli.main_list:print_packages(167): installed conda packages:
{Dist(_Dist__initd=True, channel='defaults', dist_name='wheel-0.29.0-py36_0', name='wheel', version='0.29.0', build_string='py36_0', build_number=0, with_features_depends=None, base_url=None, platform=None), Dist(_Dist__initd=True, channel='defaults', dist_name='readline-6.2-2', name='readline', version='6.2', build_string='2', build_number=2, with_features_depends=None, base_url=None, platform=None), Dist(_Dist__initd=True, channel='defaults', dist_name='pycosat-0.6.2-py36_0', name='pycosat', version='0.6.2', build_string='py36_0', build_number=0, with_features_depends=None, base_url=None, platform=None), Dist(_Dist__initd=True, channel='defaults', dist_name='pip-9.0.1-py36_1', name='pip', version='9.0.1', build_string='py36_1', build_number=1, with_features_depends=None, base_url=None, platform=None), Dist(_Dist__initd=True, channel='defaults', dist_name='pyopenssl-16.2.0-py36_0', name='pyopenssl', version='16.2.0', build_string='py36_0', build_number=0, with_features_depends=None, base_url=None, platform=None), Dist(_Dist__initd=True, channel='defaults', dist_name='conda-env-2.6.0-0', name='conda-env', version='2.6.0', build_string='0', build_number=0, with_features_depends=None, base_url=None, platform=None), Dist(_Dist__initd=True, channel='defaults', dist_name='six-1.10.0-py36_0', name='six', version='1.10.0', build_string='py36_0', build_number=0, with_features_depends=None, base_url=None, platform=None), Dist(_Dist__initd=True, channel='defaults', dist_name='requests-2.13.0-py36_0', name='requests', version='2.13.0', build_string='py36_0', build_number=0, with_features_depends=None, base_url=None, platform=None), Dist(_Dist__initd=True, channel='defaults', dist_name='conda-4.3.11-py36_0', name='conda', version='4.3.11', build_string='py36_0', build_number=0, with_features_depends=None, base_url=None, platform=None), Dist(_Dist__initd=True, channel='defaults', dist_name='zlib-1.2.8-3', name='zlib', version='1.2.8', build_string='3', build_number=3, with_features_depends=None, base_url=None, platform=None), Dist(_Dist__initd=True, channel='defaults', dist_name='openssl-1.0.2k-1', name='openssl', version='1.0.2k', build_string='1', build_number=1, with_features_depends=None, base_url=None, platform=None), Dist(_Dist__initd=True, channel='defaults', dist_name='ruamel_yaml-0.11.14-py36_1', name='ruamel_yaml', version='0.11.14', build_string='py36_1', build_number=1, with_features_depends=None, base_url=None, platform=None), Dist(_Dist__initd=True, channel='defaults', dist_name='xz-5.2.2-1', name='xz', version='5.2.2', build_string='1', build_number=1, with_features_depends=None, base_url=None, platform=None), Dist(_Dist__initd=True, channel='defaults', dist_name='tk-8.5.18-0', name='tk', version='8.5.18', build_string='0', build_number=0, with_features_depends=None, base_url=None, platform=None), Dist(_Dist__initd=True, channel='defaults', dist_name='pycparser-2.17-py36_0', name='pycparser', version='2.17', build_string='py36_0', build_number=0, with_features_depends=None, base_url=None, platform=None), Dist(_Dist__initd=True, channel='defaults', dist_name='sqlite-3.13.0-0', name='sqlite', version='3.13.0', build_string='0', build_number=0, with_features_depends=None, base_url=None, platform=None), Dist(_Dist__initd=True, channel='defaults', dist_name='setuptools-27.2.0-py36_0', name='setuptools', version='27.2.0', build_string='py36_0', build_number=0, with_features_depends=None, base_url=None, platform=None), Dist(_Dist__initd=True, channel='defaults', dist_name='python-3.6.1-0', name='python', version='3.6.1', build_string='0', build_number=0, with_features_depends=None, base_url=None, platform=None), Dist(_Dist__initd=True, channel='defaults', dist_name='idna-2.2-py36_0', name='idna', version='2.2', build_string='py36_0', build_number=0, with_features_depends=None, base_url=None, platform=None), Dist(_Dist__initd=True, channel='defaults', dist_name='pyasn1-0.2.3-py36_0', name='pyasn1', version='0.2.3', build_string='py36_0', build_number=0, with_features_depends=None, base_url=None, platform=None), Dist(_Dist__initd=True, channel='defaults', dist_name='cryptography-1.7.1-py36_0', name='cryptography', version='1.7.1', build_string='py36_0', build_number=0, with_features_depends=None, base_url=None, platform=None), Dist(_Dist__initd=True, channel='defaults', dist_name='yaml-0.1.6-0', name='yaml', version='0.1.6', build_string='0', build_number=0, with_features_depends=None, base_url=None, platform=None), Dist(_Dist__initd=True, channel='defaults', dist_name='libffi-3.2.1-1', name='libffi', version='3.2.1', build_string='1', build_number=1, with_features_depends=None, base_url=None, platform=None), Dist(_Dist__initd=True, channel='defaults', dist_name='cffi-1.9.1-py36_0', name='cffi', version='1.9.1', build_string='py36_0', build_number=0, with_features_depends=None, base_url=None, platform=None)}

```
Successful update:
```bash
(root) $  conda update --debug --dry-run --all
DEBUG conda.core.index:fetch_index(547): channel_urls=OrderedDict([('https://repo.continuum.io/pkgs/free/linux-64', ('defaults', 0)), ('https://repo.continuum.io/pkgs/free/noarch', ('defaults', 0)), ('https://repo.continuum.io/pkgs/r/linux-64', ('defaults', 0)), ('https://repo.continuum.io/pkgs/r/noarch', ('defaults', 0)), ('https://repo.continuum.io/pkgs/pro/linux-64', ('defaults', 0)), ('https://repo.continuum.io/pkgs/pro/noarch', ('defaults', 0))])
Fetching package metadata ...DEBUG requests.packages.urllib3.util.retry:from_int(191): Converted retries value: 3 -> Retry(total=3, connect=None, read=None, redirect=None)
DEBUG conda.core.index:fetch_repodata(481): Using cached repodata for https://repo.continuum.io/pkgs/free/linux-64 at /home/maba/code/miniconda3/pkgs/cache/db552c1e.json. Timeout in 14 sec
DEBUG conda.core.index:read_pickled_repodata(376): found pickle file /home/maba/code/miniconda3/pkgs/cache/db552c1e.q
.DEBUG conda.core.index:fetch_repodata(481): Using cached repodata for https://repo.continuum.io/pkgs/free/noarch at /home/maba/code/miniconda3/pkgs/cache/9ca791dd.json. Timeout in 14 sec
DEBUG conda.core.index:read_pickled_repodata(376): found pickle file /home/maba/code/miniconda3/pkgs/cache/9ca791dd.q
.DEBUG conda.core.index:fetch_repodata(481): Using cached repodata for https://repo.continuum.io/pkgs/r/linux-64 at /home/maba/code/miniconda3/pkgs/cache/6bfb14bd.json. Timeout in 24 sec
DEBUG conda.core.index:read_pickled_repodata(376): found pickle file /home/maba/code/miniconda3/pkgs/cache/6bfb14bd.q
.DEBUG conda.core.index:fetch_repodata(481): Using cached repodata for https://repo.continuum.io/pkgs/r/noarch at /home/maba/code/miniconda3/pkgs/cache/716b88c0.json. Timeout in 24 sec
.DEBUG conda.core.index:fetch_repodata(481): Using cached repodata for https://repo.continuum.io/pkgs/pro/linux-64 at /home/maba/code/miniconda3/pkgs/cache/8db5e652.json. Timeout in 24 sec
DEBUG conda.core.index:read_pickled_repodata(376): found pickle file /home/maba/code/miniconda3/pkgs/cache/8db5e652.q
.DEBUG conda.core.index:fetch_repodata(481): Using cached repodata for https://repo.continuum.io/pkgs/pro/noarch at /home/maba/code/miniconda3/pkgs/cache/08159f17.json. Timeout in 24 sec
.
DEBUG conda.plan:add_defaults_to_specs(391): H0 specs=['ruamel_yaml', 'idna', 'readline', 'pip', 'pycparser', 'yaml', 'requests', 'xz', 'pycosat', 'cffi', 'conda', 'wheel', 'openssl', 'pyasn1', 'pyopenssl', 'cryptography', 'python', 'conda-env', 'sqlite', 'libffi', 'six', 'setuptools', 'tk', 'zlib']
DEBUG conda.plan:add_defaults_to_specs(406): H2 python True
DEBUG conda.plan:add_defaults_to_specs(422): H3 python
DEBUG conda.plan:add_defaults_to_specs(406): H2 lua False
DEBUG conda.plan:add_defaults_to_specs(411): H2A lua
DEBUG conda.plan:add_defaults_to_specs(437): HF specs=['ruamel_yaml', 'idna', 'readline', 'pip', 'pycparser', 'yaml', 'requests', 'xz', 'pycosat', 'cffi', 'conda', 'wheel', 'openssl', 'pyasn1', 'pyopenssl', 'cryptography', 'python', 'conda-env', 'sqlite', 'libffi', 'six', 'setuptools', 'tk', 'zlib', 'python 3.6* (target=defaults::python-3.6.1-0)']
DEBUG conda.plan:augment_specs(711): Pinned specs=[]
DEBUG conda.resolve:install_specs(794): Checking satisfiability of current install
DEBUG conda.resolve:bad_installed(744): Checking if the current environment is consistent
Solving package specifications: DEBUG conda.resolve:solve(849): Solving for [MatchSpec('ruamel_yaml'), MatchSpec('idna'), MatchSpec('readline'), MatchSpec('pip'), MatchSpec('pycparser'), MatchSpec('yaml'), MatchSpec('requests'), MatchSpec('xz'), MatchSpec('pycosat'), MatchSpec('cffi'), MatchSpec('conda'), MatchSpec('wheel'), MatchSpec('openssl'), MatchSpec('pyasn1'), MatchSpec('pyopenssl'), MatchSpec('cryptography'), MatchSpec('python'), MatchSpec('conda-env'), MatchSpec('sqlite'), MatchSpec('libffi'), MatchSpec('six'), MatchSpec('setuptools'), MatchSpec('tk'), MatchSpec('zlib'), MatchSpec('python 3.6* (target=defaults::python-3.6.1-0)')]
DEBUG conda.resolve:get_reduced_index(380): Retrieving packages for: [MatchSpec('ruamel_yaml'), MatchSpec('idna'), MatchSpec('readline'), MatchSpec('pip'), MatchSpec('pycparser'), MatchSpec('yaml'), MatchSpec('requests'), MatchSpec('xz'), MatchSpec('pycosat'), MatchSpec('cffi'), MatchSpec('conda'), MatchSpec('wheel'), MatchSpec('openssl'), MatchSpec('pyasn1'), MatchSpec('pyopenssl'), MatchSpec('cryptography'), MatchSpec('python'), MatchSpec('conda-env'), MatchSpec('sqlite'), MatchSpec('libffi'), MatchSpec('six'), MatchSpec('setuptools'), MatchSpec('tk'), MatchSpec('zlib'), MatchSpec('python 3.6* (target=defaults::python-3.6.1-0)')]
DEBUG conda.resolve:filter_group(405): python: pruned from 82 -> 2
DEBUG conda.resolve:filter_group(405): openssl: pruned from 22 -> 10
DEBUG conda.resolve:filter_group(405): sqlite: pruned from 5 -> 1
DEBUG conda.resolve:filter_group(405): tk: pruned from 3 -> 1
DEBUG conda.resolve:filter_group(405): xz: pruned from 4 -> 2
DEBUG conda.resolve:filter_group(405): pip: pruned from 105 -> 1
DEBUG conda.resolve:filter_group(405): setuptools: pruned from 198 -> 1
DEBUG conda.resolve:filter_group(405): wheel: pruned from 17 -> 1
DEBUG conda.resolve:filter_group(405): six: pruned from 38 -> 1
DEBUG conda.resolve:filter_group(405): conda-env: pruned from 64 -> 1
DEBUG conda.resolve:filter_group(405): cryptography: pruned from 70 -> 1
DEBUG conda.resolve:filter_group(405): cffi: pruned from 68 -> 1
DEBUG conda.resolve:filter_group(405): libffi: pruned from 3 -> 2
DEBUG conda.resolve:filter_group(405): pycparser: pruned from 30 -> 1
DEBUG conda.resolve:filter_group(405): idna: pruned from 13 -> 2
DEBUG conda.resolve:filter_group(405): pyasn1: pruned from 20 -> 2
DEBUG conda.resolve:filter_group(405): pyopenssl: pruned from 26 -> 1
DEBUG conda.resolve:filter_group(405): conda: pruned from 378 -> 10
DEBUG conda.resolve:filter_group(405): pycosat: pruned from 17 -> 2
DEBUG conda.resolve:filter_group(405): requests: pruned from 106 -> 2
DEBUG conda.resolve:filter_group(405): ruamel_yaml: pruned from 10 -> 1
DEBUG conda.resolve:filter_group(405): yaml: pruned from 2 -> 1
.DEBUG conda.logic:minimize(485): Empty objective, trivial solution
DEBUG conda.resolve:solve(889): Package removal metric: 0
DEBUG conda.logic:minimize(557): Final peak objective: 0
DEBUG conda.logic:minimize(557): Final peak objective: 0
DEBUG conda.resolve:solve(895): Initial package version/build metrics: 0/0
DEBUG conda.logic:minimize(485): Empty objective, trivial solution
DEBUG conda.resolve:solve(900): Track feature count: 0
DEBUG conda.logic:minimize(485): Empty objective, trivial solution
DEBUG conda.resolve:solve(906): Package feature count: 0
DEBUG conda.logic:minimize(557): Final sum objective: 0
DEBUG conda.resolve:solve(911): Dependency update count: 0
DEBUG conda.logic:minimize(485): Empty objective, trivial solution
DEBUG conda.logic:minimize(485): Empty objective, trivial solution
DEBUG conda.resolve:solve(917): Additional package version/build metrics: 0/0
DEBUG conda.logic:minimize(485): Empty objective, trivial solution
DEBUG conda.resolve:solve(922): Weak dependency count: 0
DEBUG conda.resolve:solve(927): Looking for alternate solutions


Package plan for installation in environment /home/maba/code/miniconda3:

The following packages will be UPDATED:

    conda: 4.3.11-py36_0 --> 4.3.16-py36_0

```
The real difference in those two logs seems to be in the last couple debug outputs concerning `conda.logic.minimize`, i.e., `Final peak objective: 0` (failed run) vs. `Empty objective, trivial solution` (successful run):
```bash
DEBUG conda.resolve:solve(911): Dependency update count: 0
DEBUG conda.logic:minimize(557): Final peak objective: 0
DEBUG conda.logic:minimize(485): Empty objective, trivial solution
DEBUG conda.resolve:solve(917): Additional package version/build metrics: 0/0
DEBUG conda.logic:minimize(485): Empty objective, trivial solution
DEBUG conda.resolve:solve(922): Weak dependency count: 0
DEBUG conda.resolve:solve(927): Looking for alternate solutions
```
```bash
DEBUG conda.resolve:solve(911): Dependency update count: 0
DEBUG conda.logic:minimize(485): Empty objective, trivial solution
DEBUG conda.logic:minimize(485): Empty objective, trivial solution
DEBUG conda.resolve:solve(917): Additional package version/build metrics: 0/0
DEBUG conda.logic:minimize(485): Empty objective, trivial solution
DEBUG conda.resolve:solve(922): Weak dependency count: 0
DEBUG conda.resolve:solve(927): Looking for alternate solutions

**Comment by kalefranz** [comment]
What's `conda info` showing?

**Comment by kalefranz** [comment]
Oh I see.  Thanks for the report.

**Comment by ma-ba** [comment]
Upon closer inspection of the debug output, I see there is the problem:
```python
DEBUG conda.plan:add_defaults_to_specs(391): H0 specs=['readline', 'cryptography', 'idna', 'pip', 'sqlite', 'requests', 'wheel', 'pyasn1', 'yaml', 'tk', 'python', 'openssl', 'cffi', 'pycosat', 'conda-env', 'setuptools', 'pyopenssl', 'libffi', 'pycparser', 'six', 'zlib', 'ruamel_yaml', 'xz', 'conda-env']
...
DEBUG conda.plan:add_defaults_to_specs(437): HF specs=['readline', 'cryptography', 'idna', 'pip', 'sqlite', 'requests', 'wheel', 'pyasn1', 'yaml', 'tk', 'python', 'openssl', 'cffi', 'pycosat', 'conda-env', 'setuptools', 'pyopenssl', 'libffi', 'pycparser', 'six', 'zlib', 'ruamel_yaml', 'xz', 'conda-env', 'python 3.6* (target=defaults::python-3.6.1-0)']
```
Instead of having `['conda', 'conda-env']` in the specs, there is `['conda-env', 'conda-env']`.

**Comment by ma-ba** [comment]
The cause was this in `conda/plan.py`:
```python
def match_to_original_specs(str_specs, specs_for_prefix):                               
    matches_any_spec = lambda dst: next(spc for spc in str_specs if spc.startswith(dst))
```
which is already fixed via #5103! Closing, cheers.
